islandora.module

```diff
@@ -1685,8 +1690,12 @@ function islandora_object_access($op, $object, $user = NULL) {
  */
 function islandora_islandora_object_access($op, $object, $user) {
   module_load_include('inc', 'islandora', 'includes/utilities');
-
-  return islandora_namespace_accessible($object->id) && user_access($op, $user);
+  if ($object->state != 'A') {
+    return islandora_namespace_accessible($object->id) && user_access($op, $user) && user_access(ISLANDORA_ACCESS_INACTIVE_AND_DELETED_OBJECTS, $user);
```

Comment:
This is a point of contention I feel. I believe a submodule was suggested as it'd have the lowest point of impact on the existing code base. However, I feel like the use cases are as follows:

Institution A wants their users to have access to inactive/deleted objects even though they may not be directly accessible via the search results / browse interface (current behavior).

Institution B wants all their users to **not** have access to inactive/deleted objects full stop (new behavior + improvement).

Institution C wants all their users to **not** have access to inactive/deleted objects except certain groups of a user on a per role basis (new behavior + improvement).

The current behavior matches institution A's use case with this new code assuming the update hook is ran. If an administrator at institution A wants to create a new role they need to be aware of this new permission and grant it. Similarly, if there are permissions bundled together in features, or another configuration control, they would need to be updated to grant this new permission to these roles.

There is varied opinion as to whether to preserve existing functionality and user experience but the status quo in Islandora is to do so. With that in mind, this improvement could be re-rolled into a feature toggle with the existing permission. If you have any questions feel free to ping me here or in irc.